### ISSUE TITLE
`um` initialization is silent and takes a long time

### ISSUE THREAD

**Issue by wgordon17** [promptops/cli#5]
Finally updating the `um` CLI (after #3), and I ran a naked `um` to initialize it. It just sits there, staring at me, with no feedback until it finally dies with an error from #4.

This isn't a _bug_ per se, mostly just feedback that it would be nice to have _some_ kind of spinner, message, or otherwise that helps me understand what's going on, and that it's not just hanging indefinitely.

**Comment by jj-ctrlstack** [comment]
This should now give you progress bar if the number of history commands to index requires multiple batches. Thanks for reporting!

**Comment by wgordon17** [comment]
I see the new progress bar, very nice 👍 It seemed to stick on 6% for the _longest_ time, but I left it to run overnight and it eventually hit 100% ¯\_(ツ)_/¯

**Comment by jj-ctrlstack** [comment]
Hey @wgordon17 glad this eventually completed! Out of curiosity how many history entries do you currently have? I'm wondering if we should index the last 5k by default and if we find more warn that it might take longer. Thoughts?

**Comment by wgordon17** [comment]
Well, considering that this history file goes back 6 years 😅 there's 75k lines of history (`fish` history includes other metadata like timestamps, etc...so it's not _all_ just command history).

It would definitely be nice to have more configuration options, like only indexing the past ~1 year or only indexing the last ~5k commands. And then prompt to continue indexing or stop. Alternatively, perhaps an option to move indexing to a background task, while still allowing the `um` to be usable.

**Comment by jj-ctrlstack** [comment]
Got it so something like 30k commands. What you are suggesting sounds good, coming soon!

**Comment by jj-ctrlstack** [comment]
Fixed in 0.1.6. Thanks!
